src/main/java/com/google/devtools/build/lib/syntax/MethodLibrary.java

```diff
@@ -102,6 +105,19 @@ private static Object findExtreme(
     }
   }
 
+  private static Object evalKey(Object obj, Object key, Location loc, Environment env) throws EvalException, InterruptedException {
+    if (key == Runtime.NONE) {
+      return obj;
+    } else if (key instanceof BuiltinCallable) {
+      return ((BuiltinCallable) key).call(new ArrayList<>(Arrays.asList(obj)),
+              new HashMap<>(), new FuncallExpression(Identifier.of(""), ImmutableList.of()), env);
```

Comment:
The FuncallExpression parameter in the call function is bad design, and I plan to remove it. Please allocate exactly one instance and reuse it, rather than allocate memory on each call.